packages/cli/src/models/local/UnpackController.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import request from 'request-promise-native';
+import fs from 'fs-extra';
+
+import { Logger } from 'zos-lib';
+
+import stdout from '../../utils/stdout';
+import patch from '../../utils/patch';
+import cp from '../../utils/cp';
+
+const ora = patch('ora');
+const simpleGit = patch('simple-git/promise');
+
+interface KitConfig {
+  ignore: string[],
+  hooks: Object,
+}
+
+export default class UnpackController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = cp;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
+
+    try {
+      let spinner = ora(`Downloading kit from ${url}`).start();
+      const git = simpleGit(workingDirPath);
+      await git.init();
+      await git.addRemote('origin', url);
+      await git.pull('origin', 'stable');
+      spinner.succeed();
+
+      spinner = ora(`Unpacking kit`).start();
+      // always delete .git folder 
+      config.ignore.push(`.git`);
+      // delete all files/folders from ignore
+      const paths = config.ignore.map(file => path.join(workingDirPath, file));
+      const exists = await Promise.all(paths.map(path => pathExists(path)));
+      await Promise.all(exists.map((ext, i)=> ext ? remove(paths[i]) : null ).filter(Boolean));
+      
+      // run kit commands like `npm install`
+      await exec(config.hooks['post-unpack']);
+      spinner.succeed();
+
+      stdout(`Done. Amazing!`);
+      stdout(`Please, continue at ${url}`);
+    } catch(e) {
+      // TODO: remove all files from directory on fail except .zos.lock
+      e.message = `Failed to download and unpack kit from ${url}. Details: ${e.message}`;
+      throw e;
+    }
+  }
+
+  public async verifyRepo(url: string): Promise<KitConfig | never> {
+    if (!url) throw Error('A url must be provided.');
+
+    try {
+      return JSON.parse(await request(url
```

Comment:
Let's validate that the JSON file has the expected fields. Also, it'd be good to add some form of versioning to zepkits (similar to what we are doing in the zos.json files), in case we want to change the config format in the future.

Reply:
Yeah. I've skipped it as a decision to deliver MVP faster but It seems it didn't work!

Reply:
Hehe I like the idea of pushing the MVP, but versioning is like backups in that you only notice you need them when it's too late.

Or like smart contract upgrades, for that matter.

Reply:
Am I missing something or ZosPackageFile doesn't really validate `zos.json` except zosversion? I mean It doesn't check if any other fields are present or some extra fields not present? Do we have a npm package for validating JSON? If not I'll use https://www.npmjs.com/package/ajv